src/server/src/main/java/io/cassandrareaper/jmx/ClusterFacade.java

```diff
@@ -0,0 +1,674 @@
+/*
+ * Copyright 2017-2017 Spotify AB
+ * Copyright 2017-2019 The Last Pickle Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.cassandrareaper.jmx;
+
+import io.cassandrareaper.AppContext;
+import io.cassandrareaper.ReaperApplicationConfiguration.DatacenterAvailability;
+import io.cassandrareaper.ReaperException;
+import io.cassandrareaper.core.Cluster;
+import io.cassandrareaper.core.Compaction;
+import io.cassandrareaper.core.DroppedMessages;
+import io.cassandrareaper.core.GenericMetric;
+import io.cassandrareaper.core.JmxStat;
+import io.cassandrareaper.core.MetricsHistogram;
+import io.cassandrareaper.core.Node;
+import io.cassandrareaper.core.Segment;
+import io.cassandrareaper.core.Snapshot;
+import io.cassandrareaper.core.Table;
+import io.cassandrareaper.core.ThreadPoolStat;
+import io.cassandrareaper.resources.view.NodesStatus;
+import io.cassandrareaper.service.RingRange;
+
+import java.io.IOError;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.management.JMException;
+import javax.management.MalformedObjectNameException;
+import javax.management.ReflectionException;
+import javax.management.openmbean.CompositeData;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class ClusterFacade {
+  private static final Logger LOG = LoggerFactory.getLogger(ClusterFacade.class);
+  private static final String LOCALHOST = "127.0.0.1";
```

Comment:
this constant doesn't appear to be used anywhere. (it's part of the subsequent PR that's to follow, i believe.)